ios/XMTPModule.swift

```diff
@@ -1459,6 +1475,18 @@ public class XMTPModule: Module {
 		  }
 		  return await client.contacts.isGroupDenied(groupId: groupId)
 		}
+
+		OnAppBecomesActive {
+			Task {
+				try await clientsManager.reconnectAllLocalDatabaseConnections()
```

Comment:
I haven't been following all the twists and turns of the database reconnection story, but have we tried doing the reconnect lazily from inside `libxmtp`? So, when we use `store.conn()` and it gets an error because the connection was dropped it automatically attempts to reconnect.

That seems less brittle than having to race to connect before we try to use the DB

Reply:
I'm not sure it would work. The goal is to prevent writing to db when app is backgrounded.
So if there is for instance a message that gets streamed it will not be able to write to db.
If libxmtp just reconnects the db when it wants we will keep having crashes because it might go to background, but still reopen the connection to the db when the message gets streamed, then will be killed by the os when the app gets suspended.

Reply:
Currently our main issue is that the db is NOT disconnected when it should

Reply:
Hmm. I see